### PR TITLE
Add a test for swift split debug-info

### PR DESCRIPTION
**Pull request by fjricci**
This is currently only tested for C code, I think it's worth adding a swift version.

### CODE REVIEW

**Comment by fjricci** [comment]
@swift-ci please test

**Comment by fjricci** [comment]
Looks like this failed CI because of "IO failure on output stream." Is disk space on the test bots a KP?

**Comment by fjricci** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
@shahmishal When you get back, sounds like some bots need to be looked at.

**Comment by fjricci** [comment]
@swift-ci please test

**Comment by fjricci** [comment]
ping

File: packages/Python/lldbsuite/test/make/Makefile.rules [hunk 1]
```diff
@@ -576,7 +576,7 @@ endif
 else
 ifeq "$(SPLIT_DEBUG_SYMBOLS)" "YES"
 	$(OBJCOPY) --only-keep-debug "$(EXE)" "$(DSYM)"
-	$(OBJCOPY) --strip-debug --add-gnu-debuglink="$(DSYM)" "$(EXE)" "$(EXE)"
+	$(OBJCOPY) --remove-section .gnu_debuglink --strip-debug --add-gnu-debuglink="$(DSYM)" "$(EXE)" "$(EXE)"
```
Comment (jimingham) [inline packages/Python/lldbsuite/test/make/Makefile.rules:579]: Why do you have to remove the .gnu_debuglink section for Swift when the equivalent test didn't have to for C.

**Review by jimingham — COMMENTED** [review]
The test looks fine on it's own.  But I don't know why you changed the build rules, can you say a word about that?

**Comment by dcci** [comment]
@fjricci this is good, I think we can go ahead and merge this once you reply to Jim. I'll take care of the merge.

Thanks!

**Review by dcci — APPROVED** [review]
LGTM once Jim's comments are addressed. Please ping me and I'll merge this.

**Comment by fjricci** [comment]
I'm out of the office until next week, will do some digging when I get back. But I believe that removing before adding is fairly standard even in C/C++ codebases. Will try to find a better explanation though.

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
I'm going to test this one, and merge if it works. I think it's fine.

**Comment by dcci** [comment]
Thanks for your contribution @fjricci

**Comment by fjricci** [comment]
Those failures look unrelated (since they're compile errors). Do I need to rebase, or just wait a bit and re-test?

**Comment by dcci** [comment]
Please wait a bit and rebase/retest. Thanks!

**Comment by dcci** [comment]
@swift-ci please test

**Comment by fjricci** [comment]
@swift-ci please test

**Comment by adrian-prantl** [comment]
I'll do that, thanks!

**Comment by dcci** [comment]
Francis, we had to revert this because it broke the bot (see log attached).
Feel free to update the PR and we'll take another look

**Comment by fjricci** [comment]
Sounds good. I'll finally have access to my swift VM again on Monday, so I'll take a look then.

**Pull request merged**